### PR TITLE
[v8] Access request compatibility for servers without v2 api

### PR DESCRIPTION
**Pull request by xacrimon**
Backport #13373 to branch/v8

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

**Review by codingllama — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Review by zmb3 — CHANGES_REQUESTED** [review]
Removing approval until we verify the fix.

**Review by r0mant — CHANGES_REQUESTED** [review]
This fix doesn't appear to work, same error from CI when publishing teleport-plugins as before:

```
Original Error: *trace.NotImplementedError unknown method GetAccessRequestsV2 for service proto.AuthService
Stack Trace:
	/go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20220613212417-f490ba2be6ed/client/client.go:765 github.com/gravitational/teleport/api/client.(*Client).GetAccessRequests
	/workspace/access/jira/app.go:319 github.com/gravitational/teleport-plugins/access/jira.(*App).onJiraWebhook
	/workspace/access/jira/webhook_server.go:123 github.com/gravitational/teleport-plugins/access/jira.(*WebhookServer).processWebhook
	/go/pkg/mod/github.com/rw-access/httprouter@v1.3.1-0.20210321233808-98e93175c124/router.go:474 github.com/julienschmidt/httprouter.(*Router).ServeHTTP
	/usr/local/go/src/net/http/server.go:2879 net/http.serverHandler.ServeHTTP
	/usr/local/go/src/net/http/server.go:1930 net/http.(*conn).serve
	/usr/local/go/src/runtime/asm_amd64.s:1581 runtime.goexit
User Message: unknown method GetAccessRequestsV2 for service proto.AuthService
```

File: api/client/client_test.go [hunk 1]
```diff
@@ -574,3 +574,47 @@ func TestGetResources(t *testing.T) {
...
+			addr:                           l.Addr().String(),
+			grpc:                           grpc.NewServer(),
+			UnimplementedAuthServiceServer: &proto.UnimplementedAuthServiceServer{},
+		},
+	}
+	proto.RegisterAuthServiceServer(m.grpc, m)
+	t.Cleanup(m.grpc.Stop)
+
+	go func() {
+		require.NoError(t, m.grpc.Serve(l))
```
Comment (codingllama) [inline api/client/client_test.go:611]:
```suggestion
		assert.NoError(t, m.grpc.Serve(l))
```

Avoid "t.Fatal"-ing in goroutines.

File: api/client/client_test.go [hunk 2]
```diff
@@ -574,3 +574,47 @@ func TestGetResources(t *testing.T) {
...
+	}
+
+	return &proto.AccessRequests{
+		AccessRequests: []*types.AccessRequestV3{req.(*types.AccessRequestV3)},
+	}, nil
+}
+
+// TestAccessRequestDowngrade tests that the client will downgrade to the non stream API for fetching access requests
+// if the stream API is not available.
+func TestAccessRequestDowngrade(t *testing.T) {
```
Comment (codingllama) [inline api/client/client_test.go:595]: Nice, can we forward-port this test?

Comment (codingllama) [reply api/client/client_test.go:595]: Nevermind, already done.

**Review by codingllama — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Review by marcoandredinis — APPROVED** [review]

**Comment by xacrimon** [comment]
fyi @r0mant you'll need to approve this in order to merge

**Review by r0mant — APPROVED** [review]

**Pull request merged**